### PR TITLE
initial attempt for builing infrastructure - #54

### PR DESCRIPTION
**Pull request by jGaboardi**
initial attempt for builing infrastructure - #54

* first small set of tests passing.
* As noted in `.github/workflows/testing.yml`, pathing and data are bit of a pain due to the "convenience" functions I wrote 🙄 . Not sure how we want to proceed with that.
* We can fill in fields in `pyproject.toml` for author, etc later

xref #38

### CODE REVIEW

File: ci/py311_simplification-latest.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: py311_simplification-latest
+channels:
+  - conda-forge
+  - nodefaults
```
Comment (martinfleis) [inline ci/py311_simplification-latest.yaml:4]: what is this?

**Review by martinfleis — APPROVED** [review]

File: ci/py311_simplification-latest.yaml (hunk 1 above)
Comment (jGaboardi) [reply ci/py311_simplification-latest.yaml:4]: Something that carried over from [`environment.yml`](https://github.com/uscuni/simplification/blob/ed070520a695e50a909906d13bf9a367d23470fd/environment.yml#L4). If I am not mistaken it was declared in @anastassiavybornova's original environment. Shall we try without?

Comment (anastassiavybornova) [reply ci/py311_simplification-latest.yaml:4]: yes i think that can be dropped. i don't remember what this was for tbh (some stackexchange solution for some dependency issues i had?) and i think martin you already asked "what is this" more than once :D

**Review by anastassiavybornova — APPROVED** [review]

File: ci/py311_simplification-latest.yaml (hunk 1 above)
Comment (jGaboardi) [reply ci/py311_simplification-latest.yaml:4]: OK. I'll update this PR + `environment.yml` and repush.

**Comment by jGaboardi** [comment]
@anastassiavybornova @martinfleis 

I removed the `nodefaults` channel from all environments and added a development CI environment.

**Review by anastassiavybornova — APPROVED** [review]

**Pull request merged**